$OpenBSD: patch-aclocal_m4,v 1.1 2014/05/27 20:43:50 kili Exp $

- Fix version comparision for large version numbers.

- Allow older (buggy?) metapost.

--- aclocal.m4.orig	Mon Mar 17 16:29:16 2014
+++ aclocal.m4	Thu May 22 20:15:38 2014
@@ -116,7 +116,7 @@ AC_DEFUN(STEPMAKE_CHECK_VERSION, [
     num=`STEPMAKE_NUMERIC_VERSION($ver)`
     req=`STEPMAKE_NUMERIC_VERSION($3)`
     AC_MSG_RESULT([$ver])
-    if test "$num" -lt "$req"; then
+    if test $(dc -e "[[1]] sa 0 $num $req <a p") = 0; then
 	STEPMAKE_ADD_ENTRY($2, ["$r >= $3 (installed: $ver)"])
     fi
     vervar="`echo $1 | tr '[a-z]' '[A-Z]'`_VERSION"
@@ -1171,7 +1171,7 @@ AC_DEFUN(STEPMAKE_TEXMF, [
 	ver=`STEPMAKE_GET_VERSION($METAPOST)`
 	num=`STEPMAKE_NUMERIC_VERSION($ver)`
 	# Avoid buggy metapost versions: 1.600 <= x < 1.803
-	if test "$num" -ge "1600000" -a "$num" -lt "1803000"; then
+	if test "$num" -ge "1600000" -a "$num" -lt "1801000"; then
             STEPMAKE_ADD_ENTRY($1, ["mpost (due to a bug in metapost, versions 1.600 <= x < 1.803 are not supported; installed: $ver)"])
 	fi
     fi
